src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -3021,6 +3021,70 @@ QgsProcessingParameterDefinition *QgsProcessingParameterBoolean::clone() const
   return new QgsProcessingParameterBoolean( *this );
 }
 
+QColor QgsProcessingParameterDefinition::modelColor() const
+{
+  QgsProcessingParameterType *paramType = QgsApplication::processingRegistry()->parameterType( type() );
+  if ( paramType )
+  {
+    return paramType->modelColor();
+  }
+
+  return QgsProcessingParameterType::defaultModelColor();
+}
+
+QString QgsProcessingParameterDefinition::userFriendlyString( const QVariant &value ) const
+{
+  if ( value.userType() == qMetaTypeId<QgsPointXY>() )
```

Comment:
All these userFriendlyString implementations should start by checking:

```suggestion
   if ( QgsVariantUtils::isNull( value ) )
     return QString();

  if ( value.userType() == qMetaTypeId<QgsPointXY>() )
```